pkg/apis/core/validation/validation.go

```diff
@@ -6974,3 +7047,264 @@ func sameLoadBalancerClass(oldService, service *core.Service) bool {
+	allErrs = append(allErrs, validatePodSchedulingSpec(&resourceClaim.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+func validatePodSchedulingSpec(spec *core.PodSchedulingSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// Checking PotentialNodes for duplicates is intentionally not done. It
+	// could be fairly expensive and the only component which normally has
+	// permissions to set this field, kube-scheduler, is a trusted
+	// component.  Also, if it gets this wrong because of a bug, then the
+	// effect is limited (same semantic).
+	if len(spec.PotentialNodes) > core.PodSchedulingNodeListMaxSize {
+		allErrs = append(allErrs, field.TooLongMaxLength(fldPath.Child("potentialNodes"), nil, core.PodSchedulingNodeListMaxSize))
+	}
+	return allErrs
+}
+
+// ValidatePodSchedulingUpdate tests if an update to PodScheduling is valid.
+func ValidatePodSchedulingUpdate(resourceClaim, oldPodScheduling *core.PodScheduling) field.ErrorList {
+	allErrs := ValidateObjectMetaUpdate(&resourceClaim.ObjectMeta, &oldPodScheduling.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidatePodScheduling(resourceClaim)...)
+	return allErrs
+}
+
+// ValidatePodSchedulingStatusUpdate tests if an update to the status of a PodScheduling is valid.
+func ValidatePodSchedulingStatusUpdate(resourceClaim, oldPodScheduling *core.PodScheduling) field.ErrorList {
+	allErrs := ValidateObjectMetaUpdate(&resourceClaim.ObjectMeta, &oldPodScheduling.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, validatePodSchedulingStatus(&resourceClaim.Status, field.NewPath("status"))...)
+	return allErrs
+}
+
+func validatePodSchedulingStatus(status *core.PodSchedulingStatus, fldPath *field.Path) field.ErrorList {
+	return validatePodSchedulingClaims(status.Claims, fldPath.Child("claims"))
+}
+
+func validatePodSchedulingClaims(claimStatuses []core.ResourceClaimSchedulingStatus, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	podResourceClaimNames := sets.NewString()
+	for i, claimStatus := range claimStatuses {
+		allErrs = append(allErrs, validatePodSchedulingClaim(claimStatus, fldPath.Index(i))...)
+		if podResourceClaimNames.Has(claimStatus.PodResourceClaimName) {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Index(i), claimStatus))
+		} else {
+			podResourceClaimNames.Insert(claimStatus.PodResourceClaimName)
+		}
+	}
+	return allErrs
+}
+
+func validatePodSchedulingClaim(claim core.ResourceClaimSchedulingStatus, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// Checking UnsuitableNodes for duplicates is intentionally not done. It
+	// could be fairly expensive and if a resource driver gets this wrong,
+	// then it is only going to have a negative effect for the pods relying
+	// on this driver.
+	if len(claim.UnsuitableNodes) > core.PodSchedulingNodeListMaxSize {
```

Comment:
This is intentionally not validating the set aspect because it would be expensive and clients which have permission to write this field are trusted.

Same with node names. If a trusted client puts some string here that isn't a valid node name, nothing bad happens.

If we do all of that additional validation, we don't gain anything and just spend more CPU cycles during updates.